examples/simulator-lang-rps-2-rps/client.mjs

```diff
@@ -1,86 +1,44 @@
-  await (await alice.getNextAction()).resolve(0);
-  // Alice's hand (0) is published
-  await (await consensus.getNextAction()).resolve(0);
-  // Alice observes that her hand is published
-  await (await alice.getNextAction()).resolve();
-  // Bob observes that Alice's hand is published
-  await (await bob.getNextAction()).resolve();
-  // Bob interactively gets his hand (1)
-  await (await bob.getNextAction()).resolve(1);
-  // Bob's hand (1) is published
-  await (await consensus.getNextAction()).resolve(1);
-  // Alice observes that Bob's hand is published
-  await (await alice.getNextAction()).resolve();
-  // Bob observes that his hand is published
-  await (await bob.getNextAction()).resolve();
-  // Alice's program has run to completion
-  let r = await alice.getStatus();
-  console.log(r);
-  assert.equal(r,"Done");
-  let store = await alice.getStore();
-  // Alice observes that her hand is published
-  store.getVar('handAlice').assertVar('V_UInt',0);
-  // Alice observes Bob's hand
-  store.getVar('handBob').assertVar('V_UInt',1);
-  // B wins
-  store.getVar('outcome').assertVar('V_UInt',0);
-
-  console.log("Imperative Scenario Done ########################");
+  const alice = is.participants.Alice;
+  const bob = is.participants.Bob;
+  const consensus = is.consensus;
+  const [, a] = await alice.init();
+  const [s, b] = await bob.init();
 
-  // test functionally
-  const fs = new lang.FunctionalScenario();
-  await fs.init();
-  pi = await fs.pingServer();
-  console.log(fs);
-  alice = fs.participants.Alice;
-  bob = fs.participants.Bob;
-  consensus = fs.consensus;
-  // init Alice
-  await alice.init();
-  // init Bob
-  await bob.init();
-  // Alice interactively gets her hand (0)
-  let s = await (await alice.getNextAction()).resolve(0);
-  // Alice's hand (0) is published
-  s = await (await s.who(consensus).getNextAction()).resolve(0);
-  // Alice observes that her hand is published
-  s = await (await s.who(alice).getNextAction()).resolve();
-  // Bob observes that Alice's hand is published
-  s = await (await s.who(bob).getNextAction()).resolve();
-  // Bob interactively gets his hand (1)
-  s = await (await s.who(bob).getNextAction()).resolve(1);
-  // Bob's hand (1) is published
-  s = await (await s.who(consensus).getNextAction()).resolve(1);
-  // Alice observes that Bob's hand is published
-  s = await (await s.who(alice).getNextAction()).resolve();
-  // Bob observes that his hand is published
-  s = await (await s.who(bob).getNextAction()).resolve();
-  r = await s.who(alice).getStatus();
-  console.log(r);
-  assert.equal(r,"Done");
-  console.log("Functional Scenario Done ########################");
+  for (let aHand = 0; aHand < 3; aHand++) {
+    for (let bHand = 0; bHand < 3; bHand++) {
+      const r = await impScenario(s.copy(),aHand,bHand,a,b,consensus);
+      r.assertVar('V_UInt',winner(aHand,bHand));
+    }
+  }
 
-  console.log("Testing Complete")
+  console.log("Testing Complete!!!")
 }
 
```

Comment:
This file looks great. I think it might look cleaner to not have the `impScenario` function and just inline it, but I think that it looks better as a function if you keep going from a DRAW and do it again.